contracts/zaps/ZapCurveTokenToPrincipalToken.sol

```diff
@@ -0,0 +1,458 @@
+            IVault.FundManagement({
+                sender: address(this),
+                fromInternalBalance: false,
+                recipient: payable(_info.recipient),
+                toInternalBalance: false
+            }),
+            _info.minPtAmount,
+            _info.deadline
+        );
+    }
+
+    struct ZapCurveLpOut {
+        // Address of the curvePool for which an amount of lpTokens
+        // is swapped for an amount of single root tokens
+        address curvePool;
+        // The contract address of the curve pools lpToken
+        IERC20 lpToken;
+        // This is the index of the target root we are swapping for
+        int128 rootTokenIdx;
+        // Address of the rootToken we are swapping for
+        address rootToken;
+        // This is the function signature of the curvePool's
+        // "remove_liquidity_one_coin" function which similar to the
+        // "add_liquidity" curvePool function in the zapIn, there is
+        // an inconsistent interface when interacting with curve pools
+        bytes4 funcSig;
+    }
+
+    struct ZapOutInfo {
+        // Pool id of balancer pool that is used to exchange a users
+        // amount of principal tokens
+        bytes32 balancerPoolId;
+        // Address of the principal token
+        IAsset principalToken;
+        // Amount of principal tokens the user wishes to swap for
+        uint256 principalTokenAmount;
+        // The recipient is the address the tokens which are to be swapped for
+        // will be sent to
+        address payable recipient;
+        // The minimum amount root tokens the user is expecting
+        uint256 minRootTokenAmount;
+        // Timestamp into the future for which a transaction is valid for
+        uint256 deadline;
+        // If the target root token is sourced via two curve pool swaps, then
+        // this is to be flagged as true
+        bool targetNeedsChildZap;
+    }
+
+    /// @dev Swaps an amount of curve lpTokens for a single dependent root token
+    /// from its pool
+    /// @param _zap See ZapCurveLpOut
+    /// @param _lpTokenAmount This is the amount of lpTokens we are swapping
+    /// with
+    /// @param _minRootTokenAmount This is the minimum amount of "root" tokens
+    /// the user expects to swap for. Used only in the final zap when executed
+    /// under zapOut
+    /// @param _recipient The address which the outputs tokens are to be sent
+    /// to. When there is a second zap to occur, in the first zap the recipient
+    /// should be this address
+    function _zapCurveLpOut(
+        ZapCurveLpOut memory _zap,
+        uint256 _lpTokenAmount,
+        uint256 _minRootTokenAmount,
+        address payable _recipient
+    ) internal returns (uint256 rootAmount) {
+        // Flag to detect if we are sending to recipient
+        bool transferToRecipient = address(this) != _recipient;
+
+        // Actual swap of lpToken -> rootToken
+        address(_zap.curvePool).functionCall(
```

Comment:
Can we put an example in the comments of how it gonna work? Like what will be the root token, curvePool , lpToken etc .. It will be better to understand.

Reply:
Added comment, LMK if ok